src/DataViews/PresetsDataView.cpp

```diff
@@ -196,9 +206,22 @@ void PresetsDataView::OnShowInExplorerRequested(const std::vector<int>& selectio
 }
 
 void PresetsDataView::OnDoubleClicked(int index) {
-  const PresetFile& preset = GetPreset(index);
+  PresetFile& preset = GetMutablePreset(index);
   if (app_->GetPresetLoadState(preset).state != PresetLoadState::kNotLoadable) {
-    app_->LoadPreset(preset);
+    app_->LoadPreset(preset).Then(main_thread_executor_.get(),
+                                  [&preset](ErrorMessageOr<void> result) {
```

Comment:
This `&preset` might be a problem. I dont think we can be 100% sure that the preset still exists when the closure is executed. For example: The user might have deleted the preset while its loading. (I know unlikely, but I think this will result in a dangling reference)

Reply:
I just realized, this closure might not be necessary at all, because of `OnLoadPresetComplete`

Reply:
`OnLoadPresetComplete` is only invoked when we load preset file from the main window menu.

Reply:
Ah gotcha, that makes sense!